apps/studio/components/interfaces/Storage/__tests__/DeleteBucketModal.test.tsx

```diff
@@ -0,0 +1,157 @@
+import { describe, expect, it, beforeEach, vi } from 'vitest'
+import { mockAnimationsApi } from 'jsdom-testing-mocks'
+import { screen, waitFor, fireEvent } from '@testing-library/dom'
+import userEvent from '@testing-library/user-event'
+import { useState } from 'react'
+import { MoreVertical } from 'lucide-react'
+import { faker } from '@faker-js/faker'
+
+import { addAPIMock } from 'tests/lib/msw'
+import {
+  Button,
+  DropdownMenu,
+  DropdownMenuContent,
+  DropdownMenuItem,
+  DropdownMenuTrigger,
+} from 'ui'
+import { ProjectContextProvider } from 'components/layouts/ProjectLayout/ProjectContext'
+import { Bucket } from 'data/storage/buckets-query'
+import DeleteBucketModal from '../DeleteBucketModal'
+import { render } from 'tests/helpers'
+import { routerMock } from 'tests/lib/route-mock'
+
+const bucket: Bucket = {
+  id: faker.string.uuid(),
+  name: `test`,
+  owner: faker.string.uuid(),
+  public: faker.datatype.boolean(),
+  allowed_mime_types: faker.helpers.multiple(() => faker.system.mimeType(), {
+    count: { min: 1, max: 5 },
+  }),
+  file_size_limit: faker.number.int({ min: 0, max: 25165824 }),
+  type: faker.helpers.arrayElement(['STANDARD', 'ANALYTICS', undefined]),
+  created_at: faker.date.recent().toISOString(),
+  updated_at: faker.date.recent().toISOString(),
+}
+
+const Page = ({ onClose }: { onClose: () => void }) => {
+  const [modal, setModal] = useState<string | null>(null)
+  const renderModal = () => {
+    switch (modal) {
+      case `delete`:
+        return (
+          <DeleteBucketModal
+            bucket={bucket}
+            onClose={() => {
+              setModal(null)
+              onClose()
+            }}
+          />
+        )
+      default:
+        return null
+    }
+  }
+  return (
+    <ProjectContextProvider projectRef="default">
+      <DropdownMenu>
+        <DropdownMenuTrigger asChild>
+          <Button title="Manage Bucket" type="text" icon={<MoreVertical />} />
+        </DropdownMenuTrigger>
+        <DropdownMenuContent>
+          <DropdownMenuItem onClick={() => setModal(`delete`)}>Delete</DropdownMenuItem>
+        </DropdownMenuContent>
+      </DropdownMenu>
+
+      {renderModal()}
+    </ProjectContextProvider>
+  )
+}
+
+mockAnimationsApi()
```

Comment:
note that this is needed anytime we've got `DropdownMenu` in the component tree

Reply:
plan to write up some notes there as I discover more edge cases that need special treatment!